flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/source/offset/BinlogOffset.java

```diff
@@ -232,7 +232,9 @@ public int compareTo(BinlogOffset that) {
             // compared ...
             long timestamp = this.getTimestampSec();
             long targetTimestamp = that.getTimestampSec();
-            return Long.compare(timestamp, targetTimestamp);
+            if (timestamp != 0 && targetTimestamp != 0) {
+                return Long.compare(timestamp, targetTimestamp);
```

Comment:
This change is necessary because that if the TimestampSec is 0, the assumption of `These are from different servers, and their binlog coordinates are not related` is not correct, these two BinlogOffsets could come from the same server.